tendermint/src/consensus/state.rs

```diff
@@ -20,6 +23,7 @@ pub struct State {
     pub step: i8,
 
     /// Block ID being proposed (if available)
+    #[serde(with = "tendermint_proto::serializers::optional")]
```

Comment:
@greg-szabo @thanethomson does this not handle deserialization in the case a field is null?

The existing state files have data like...

```json
{"height":"3820001","round":"0","step":2,"block_id":null}
```

But that no longer parses:

```
invalid type: null, expected struct BlockId at line 1 column 50
```

Reply:
see also: https://github.com/iqlusioninc/tmkms/pull/226